src/include/duckdb/common/serializer/serializer.hpp

```diff
@@ -38,6 +48,8 @@ class Serializer {
 	SerializationData data;
 
 public:
+	explicit Serializer(const SerializationOptions &opts) : options(opts) {
```

Comment:
Instead of enforcing this at the constructor level maybe we can use the `Set/Get` parameters to pass this down instead?